src/Datagrid/Pager.php

```diff
@@ -36,13 +38,12 @@ public function computeNbResult()
             $countQuery->setParameters($this->getParameters());
         }
 
-        $countQuery->select(sprintf(
-            'count(DISTINCT %s.%s) as cnt',
-            current($countQuery->getRootAliases()),
-            current($this->getCountColumn())
-        ));
+        $query = $countQuery->getQuery();
+        $isDistinct = $countQuery instanceof ProxyQuery ? $countQuery->isDistinct() : true;
```

Comment:
One time variable, can you move ir down and remove `$isDistinct`

Reply:
Ok